crypto/dilithium.go

```diff
@@ -0,0 +1,175 @@
+package crypto
+
+import (
+	"fmt"
+
+	"github.com/cloudflare/circl/sign/dilithium"
+	"github.com/pkg/errors"
+)
+
+type (
+	DilithiumMode string
+)
+
+func (m DilithiumMode) String() string {
+	return string(m)
+}
+
+const (
+	Dilithium2 DilithiumMode = "Dilithium2"
+	Dilithium3 DilithiumMode = "Dilithium3"
+	Dilithium5 DilithiumMode = "Dilithium5"
+)
+
+func GenerateDilithiumKeyPair(m DilithiumMode) (dilithium.PublicKey, dilithium.PrivateKey, error) {
+	mode := dilithium.ModeByName(m.String())
+	if mode == nil {
+		return nil, nil, fmt.Errorf("unsupported dilithium mode: %s", m)
+	}
+	pk, sk, err := mode.GenerateKey(nil)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "could not generate key for dilithium")
+	}
+	return pk, sk, nil
+}
+
+type DilithiumSigner struct {
+	kid  string
+	mode dilithium.Mode
+	dilithium.PublicKey
+	dilithium.PrivateKey
```

Comment:
Are you intending for the `DilithiumSigner` to inherit the behaviors of `dilithium.PublicKey` and `dilithium.PrivateKey`?

If not, I think it improves readability to have these fields explicitly named, rather than doing a struct embedding.

Reply:
behaviors...not exactly

if you see the sign method
```go
func (s *DilithiumSigner) Sign(message []byte) []byte {
	return s.mode.Sign(s.PrivateKey, message)
}
```
the behavior is defined on the `mode` not the key itself